app/controllers/items_controller.rb

```diff
@@ -412,22 +412,9 @@ def close_version
     end
   end
 
-  # TODO: this should be a method in dor-services, invoked within set_source_id
-  def self.normalize_source_id(new_source_id)
-    src_id_arr = new_source_id.split(':').map(&:strip)
-    raise 'new source_id must be of the form "source:value"' if src_id_arr.length != 2
-    src_id_arr.join(':')
-  end
-
   def source_id
-    new_id = ItemsController.normalize_source_id(params[:new_id])
-    @object.source_id = new_id
-    # TODO: the content= and content_will_change! calls belong in dor-services,
-    # for this method and other similar methods. can then clean up
-    # allow(idmd).to receive(:"ng_xml") (and "content_will_change!") in tests.
-    # getting rid of "Metadata\.content\s*=\s*"
-    @object.identityMetadata.content = @object.identityMetadata.ng_xml.to_xml
-    @object.identityMetadata.content_will_change!
```

Comment:
I agree this TODO and `@object` lines are obsolete.  The lines it references in test appear to already be gone and the setter from Dor::Services now calls `ng_xml_will_change!`.